src/main/java/com/labsynch/labseer/api/ApiExperimentController.java

```diff
@@ -1634,20 +1635,19 @@ public ResponseEntity<java.lang.String> jsonFindExperimentByNameGet(@RequestPara
 	@RequestMapping(value = "/protocol/{codeName}", method = RequestMethod.GET, headers = "Accept=application/json")
 	@ResponseBody
 	public ResponseEntity<java.lang.String> jsonFindExperimentsByProtocolCodeName(
-			@PathVariable("codeName") String codeName, @RequestParam(value = "with", required = false) String with) {
+			@PathVariable("codeName") String codeName, @RequestParam(value = "projects", required = false) List<String> projects, @RequestParam(value = "with", required = false) String with) {
```

Comment:
New projects query param and removed most of the logic from the api controller an moved it to a new service.